### ISSUE TITLE
Compiler Error - "Command failed due to signal: Segmentation fault: 11"

### ISSUE THREAD

**Issue by mobilinked** [ReactKit/SwiftTask#86]
Using XCode9 GM to compile SwiftTask swift3 branch with "swift version to 3.2", error as the topic

**Comment by yuvalt** [comment]
I've spent a few hours to figure it out. If you comment the line: 

            self?._machine.removeProgressTupleHandler(token)

from SwiftTask.swift, it will not break. Obviously not a fix.

**Comment by mankusti** [comment]
This is the workaround I'm using (replace the method starting at line 365):

    public func progress<C: Canceller>(_ canceller: inout C?, _ progressClosure: @escaping (ProgressTuple) -> Void) -> Self
    {
        var token: _HandlerToken? = nil
        self._machine.addProgressTupleHandler(&token, progressClosure)
        let theToken = token
        
        canceller = C { [weak self] in
            self?._machine.removeProgressTupleHandler(theToken)
        }
        
        return self
    }

This workaround is based on the discussion at: https://bugs.swift.org/browse/SR-293

**Comment by ghugues** [comment]
For anyone interested in an automated solution using cocoapods, add this at the bottom of your Podfile (based on @mankusti's comment)

```
post_install do |installer|

	# Use Swift 3.2 for dependencies that still don't support Swift 4
	installer.pods_project.targets.each do |target|
		if ['SwiftTask'].include? target.name
			target.build_configurations.each do |config|
				config.build_settings['SWIFT_VERSION'] = '3.2'
			end
		end
	end

	# Patch SwiftTask to compile with Xcode 9 (see https://github.com/ReactKit/SwiftTask/issues/86#issuecomment-329483570)
	%x(patch Pods/SwiftTask/SwiftTask/SwiftTask.swift <<< '
		@@ -366,9 +366,10 @@
		     {
		         var token: _HandlerToken? = nil
		         self._machine.addProgressTupleHandler(&token, progressClosure)
		+        let theToken = token
		         
		         canceller = C { [weak self] in
		-            self?._machine.removeProgressTupleHandler(token)
		+            self?._machine.removeProgressTupleHandler(theToken)
		         }
		         
		         return self
	')
end
```

**Comment by mobilinked** [comment]
Great!

**Comment by Visput** [comment]
Seems like ReactKit stopped supporting this project. That's weird.

**Comment by inamiy** [comment]
Hi, sorry for late response.
This fix is elegantly done in #87, so I will close this issue.
